spring-integration-core/src/main/java/org/springframework/integration/aggregator/CorrelatingMessageBarrier.java

```diff
@@ -118,7 +120,7 @@ private Object getLock(Object correlationKey) {
 						Iterator<Message<?>> messages = group.getMessages().iterator();
 						if (messages.hasNext()) {
 							nextMessage = messages.next();
-							store.removeMessageFromGroup(key, nextMessage);
+							this.store.removeMessagesFromGroup(key, Collections.<Message<?>>singletonList(nextMessage));
```

Comment:
Would you mind adding an overloaded version for `removeMessagesFromGroup` with vararg avoid this extra collection creation just for single message ?